example/example.c

```diff
@@ -0,0 +1,20 @@
+#include "example.h"
+
+#define FUNCTION_INDEX_GROUP_EXAMPLE_SIZE 2
+
+static void **function_index_group_example;
+
+void set_function_index_group_example(void **functions) {
+    function_index_group_example = functions;
+}
+
+
+int add(int a, int b) {
+    return ((int (*)(int, int))(function_index_group_example[0]))(a, b);
+}
+
+
+void print_factorial(int n) {
+    ((void (*)(int))(function_index_group_example[1]))(n);
```

Comment:
Ditto here.

Alternatively, you could declare a `typedef void (*print_factorial_ptr)(int)` and use `((print_factorial_ptr) function_index_froup_example)[1](n)` here.

Also recall that when C declarations get hairy, you can always resort to cdecl ([here](http://www.ibiblio.org/pub/linux/devel/lang/c/cdecl-2.5.tar.gz) or [here](https://cdecl.org/)) to help you make sense of it (but do use `typedef`s to avoid the need in your own code).